src/Couchbase.Lite.Shared/API/Query/IIndexable.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Couchbase.Lite.Internal.Query;
+using JetBrains.Annotations;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Couchbase.Lite.Query
+{
+    public interface IIndexable
+    {
+        /// <summary>
+        /// Gets a list of index names that are present in the database
+        /// </summary>
+        /// <returns>The list of created index names</returns>
+        [NotNull]
+        [ItemNotNull]
+        IList<string> GetIndexes();
```

Comment:
Probably can be read only list

Reply:
Database class's GetIndexes() is also implementing this interface and if changes it will break API.